lib/puppet/pops/loader/loader_paths.rb

```diff
@@ -230,6 +249,14 @@ def valid_name?(typed_name)
       # TODO: Remove when PE has proper namespace handling
       typed_name.name_parts.size <= 2
     end
+
+    def valid_path?(path)
+      path.start_with?(generic_path) && is_task_name?(File.basename(path, '.*')) && !FORBIDDEN_EXTENSIONS.any? { |ext| path.end_with?(ext) }
+    end
+
+    def is_task_name?(name)
+      !!(name =~ /^[a-z][a-z0-9_]*$/)
```

Comment:
"!!" ?

Reply:
This is the simplest way to guarantee the return of a boolean `true` or `false` based on the outcome of the regexp (which is `nil` or an integer).

Reply:
I figured there was a reason. Cool